src/core/Akka/Actor/ActorPath.cs

```diff
@@ -500,36 +581,45 @@ private static bool TryParseAddress(string path, out Address address, out ReadOn
         /// <summary>
         /// Joins this instance.
         /// </summary>
+        /// <param name="prefix">the address or empty</param>
         /// <returns> System.String. </returns>
-        private string Join()
+        private string Join(ReadOnlySpan<char> prefix)
         {
-            if (this is RootActorPath)
-                return "/";
-
-            // Resolve length of final string
-            var totalLength = 0;
-            var p = this;
-            while (!(p is RootActorPath))
+            if (_depth == 0)
             {
-                totalLength += p.Name.Length + 1;
-                p = p.Parent;
+                Span<char> buffer = stackalloc char[prefix.Length + 1];
+                prefix.CopyTo(buffer);
+                buffer[buffer.Length - 1] = '/';
+                return buffer.ToString();
             }
-
-            // Concatenate segments (in reverse order) into buffer with '/' prefixes
-            char[] buffer = new char[totalLength];
-            int offset = buffer.Length;
-            p = this;
-            while (!(p is RootActorPath))
+            else
             {
-                offset -= p.Name.Length + 1;
-                buffer[offset] = '/';
+                // Resolve length of final string
+                var totalLength = prefix.Length;
+                var p = this;
+                while (p._depth > 0)
+                {
+                    totalLength += p._name.Length + 1;
+                    p = p.Parent;
+                }
 
-                p.Name.CopyTo(0, buffer, offset + 1, p.Name.Length);
+                // Concatenate segments (in reverse order) into buffer with '/' prefixes
+                Span<char> buffer = stackalloc char[totalLength];
```

Comment:
I still worry here whether a path can be long enough that we should -not- be allocating on the stack.

This is a bigger concern on MacOS where Stack sizes can be small.

Consider:

```
Span<char> buffer = totalLength <256 ? stackalloc char[totalLength] : new char[totalLength]; // pick a good number here.
```

Reply:
sry, completely forgot, 1024 is a good number
I will fix it and make a unit test for it.

Reply:
On windows 10MB stackallock is no issue.
The stackallock could be removed, when we have `String.Create()`